tests/smoke/test_newsrec_model.py

```diff
@@ -11,97 +11,215 @@
 from reco_utils.recommender.newsrec.models.naml import NAMLModel
 from reco_utils.recommender.newsrec.models.lstur import LSTURModel
 from reco_utils.recommender.newsrec.models.npa import NPAModel
-from reco_utils.recommender.newsrec.io.news_iterator import NewsIterator
-from reco_utils.recommender.newsrec.io.naml_iterator import NAMLIterator
+from reco_utils.recommender.newsrec.io.mind_iterator import MINDIterator
+from reco_utils.recommender.newsrec.io.mind_all_iterator import MINDAllIterator
 
 
 @pytest.mark.smoke
 @pytest.mark.gpu
 def test_model_nrms(tmp):
-    yaml_file = os.path.join(tmp, "nrms.yaml")
-    train_file = os.path.join(tmp, "train.txt")
-    valid_file = os.path.join(tmp, "test.txt")
-    wordEmb_file = os.path.join(tmp, "embedding.npy")
-
+    train_news_file = os.path.join(tmp, "train", r"news.tsv")
+    train_behaviors_file = os.path.join(tmp, "train", r"behaviors.tsv")
+    valid_news_file = os.path.join(tmp, "valid", r"news.tsv")
+    valid_behaviors_file = os.path.join(tmp, "valid", r"behaviors.tsv")
+    wordEmb_file = os.path.join(tmp, "utils", "embedding.npy")
+    userDict_file = os.path.join(tmp, "utils", "uid2index.pkl")
+    wordDict_file = os.path.join(tmp, "utils", "word_dict.pkl")
+    yaml_file = os.path.join(tmp, "utils", r"nrms.yaml")
+
+    if not os.path.exists(train_news_file):
```

Comment:
I believe `tmp` is the function level, meaning a temp folder will be generated for each test function and thus the data will be downloaded anyway (as path.exists will return False). We have couple of tests that use the same dataset, and this dataset will be downloaded multiple times.

We probably can create module-level tmp folder and pass that so that a suite of tests in the same module can share the same file if necessary.

Reply:
in this case, in fact, maybe we should add a fixture that downloaded all the data once and then that to be used by all the other functions. I agree that we should have another tmp for the context module, and maybe another one for system

Reply:
do you want to create an issue for that and handle separately from this PR?

Reply:
yes, let me create it

Reply:
https://github.com/microsoft/recommenders/issues/1169